File: code/13-Java2Python/python/runner/runner_guess_game.py
from domain.guess_game import GuessGame
from domain.number_guesser import NumberGuesser
from domain.secret_number_generator import SecretNumberGenerator
from valueobjects.interval import Interval


def main() -> None:
    """Wires up and plays a single GuessGame."""
    start_interval = Interval.of(1, 100)
    generator = SecretNumberGenerator.create()
    guesser = NumberGuesser.create()
    game = GuessGame.of(start_interval, generator, guesser)
    result = game.play()
    print(result)


if __name__ == "__main__":
    main()

File: code/12-JavaCodeLoop/JavaTDDLoop/src/main/java/com/example/runner/RunnerGuessGame.java
package com.example.runner;

import com.example.valueobjects.GameResult;
import com.example.domain.GuessGame;
import com.example.domain.NumberGuesser;
import com.example.domain.SecretNumberGenerator;
import com.example.valueobjects.Interval;

/**
 * Entry point that wires up and plays a single {@link GuessGame}.
 */
public final class RunnerGuessGame {

    private RunnerGuessGame() {
    }

    public static void main(String[] args) {
        var startInterval = Interval.of(1, 100);
        var generator = SecretNumberGenerator.create();
        var guesser = NumberGuesser.create();
        var game = GuessGame.of(startInterval, generator, guesser);
        var result = game.play();
        System.out.println(result);
    }
}
